pkg/kubectl-argo-rollouts/cmd/get/get_rollout.go

```diff
@@ -0,0 +1,242 @@
+
+	fmt.Fprintf(o.Out, "\n")
+	o.PrintRolloutTree(roInfo)
+}
+
+func (o *GetOptions) PrintRolloutTree(roInfo *info.RolloutInfo) {
+	w := ansiterm.NewTabWriter(o.Out, 0, 0, 2, ' ', 0)
+	o.PrintHeader(w)
+	fmt.Fprintf(w, "%s %s\t%s\t%s %s\t%s\t%v\n", IconRollout, roInfo.Name, "Rollout", o.colorize(roInfo.Icon), roInfo.Status, roInfo.Age(), "")
+	revisions := roInfo.Revisions()
+	for i, rev := range revisions {
+		isLast := i == len(revisions)-1
+		var prefix, subpfx string
+		if !isLast {
+			prefix = "├──"
+			subpfx = "│  "
+		} else {
+			prefix = "└──"
+			subpfx = "   "
+		}
+		o.PrintRevision(w, roInfo, rev, prefix, subpfx)
+	}
+	_ = w.Flush()
+}
+
+func (o *GetOptions) PrintRevision(w io.Writer, roInfo *info.RolloutInfo, revision int, prefix string, subpfx string) {
+	name := fmt.Sprintf("revision:%d", revision)
+	fmt.Fprintf(w, "%s%s %s\t%s\t%s %s\t%s\t%v\n", prefix, IconRevision, name, "", "", "", "", "")
+	replicaSets := roInfo.ReplicaSetsByRevision(revision)
+	experiments := roInfo.ExperimentsByRevision(revision)
+	analysisRuns := roInfo.AnalysisRunsByRevision(revision)
+	total := len(replicaSets) + len(experiments) + len(analysisRuns)
+	curr := 0
+
+	getPrefixes := func() (string, string) {
+		isLast := curr == total-1
+		curr++
+		var childPrefix, childSubpfx string
+		if !isLast {
+			childPrefix = subpfx + "├──"
+			childSubpfx = subpfx + "│  "
+		} else {
+			childPrefix = subpfx + "└──"
+			childSubpfx = subpfx + "   "
+		}
+		return childPrefix, childSubpfx
+	}
+
+	for _, rsInfo := range replicaSets {
+		childPrefix, childSubpfx := getPrefixes()
+		o.PrintReplicaSetInfo(w, rsInfo, childPrefix, childSubpfx)
+	}
+	for _, expInfo := range experiments {
+		childPrefix, childSubpfx := getPrefixes()
+		o.PrintExperimentInfo(w, expInfo, childPrefix, childSubpfx)
+	}
+	for _, arInfo := range analysisRuns {
+		childPrefix, childSubpfx := getPrefixes()
+		o.PrintAnalysisRunInfo(w, arInfo, childPrefix, childSubpfx)
+	}
+}
+
+func (o *GetOptions) PrintReplicaSetInfo(w io.Writer, rsInfo info.ReplicaSetInfo, prefix string, subpfx string) {
+	infoCols := []string{}
+	name := rsInfo.Name
+	if rsInfo.Stable {
+		infoCols = append(infoCols, o.colorize(info.InfoTagStable))
+		name = o.ansiFormat(name, FgGreen)
+	} else if rsInfo.Canary {
+		infoCols = append(infoCols, o.colorize(info.InfoTagCanary))
+		name = o.ansiFormat(name, FgYellow)
+	} else if rsInfo.Active {
+		infoCols = append(infoCols, o.colorize(info.InfoTagActive))
+		name = o.ansiFormat(name, FgGreen)
+	} else if rsInfo.Preview {
+		infoCols = append(infoCols, o.colorize(info.InfoTagPreview))
+		name = o.ansiFormat(name, FgBlue)
+	}
+	fmt.Fprintf(w, "%s%s %s\t%s\t%s %s\t%s\t%v\n", prefix, IconReplicaSet, name, "ReplicaSet", o.colorize(rsInfo.Icon), rsInfo.Status, rsInfo.Age(), strings.Join(infoCols, ","))
+	for i, podInfo := range rsInfo.Pods {
+		fmt.Fprintf(w, subpfx)
+		isLast := i == len(rsInfo.Pods)-1
+		var podPrefix string
+		if !isLast {
```

Comment:
Weren't you going to make this into a helper?

Reply:
I made it an local function, because in the other case, the `isLast` is calculated slightly differently, but let me see if i can make it generic to use everywhere.